sawbuck_app/src/views/asset_list.js

```diff
@@ -0,0 +1,72 @@
+/**
+ * Copyright 2017 Intel Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * ----------------------------------------------------------------------------
+ */
+'use strict'
+
+const m = require('mithril')
+const _ = require('lodash')
+
+const layout = require('../components/layout')
+const api = require('../services/api')
+
+const offerButton = (name, key = 'source') => {
+  const label = key === 'target' ? 'Request' : 'Offer'
+  const onclick = () => console.log(`Offering ${name} as ${key}...`)
+
+  return m('button.btn.btn-outline-primary.mr-3', { onclick }, label)
+}
+
+const assetRow = asset => {
+  return m('.row.mb-5', [
+    m('.col-md-8', [
+      layout.row(m('a.h5', {
+        href: `/assets/${asset.name}`,
```

Comment:
Is asset name something that is URL safe?

Reply:
Typically yes. And for the purposes of this demo (and this UI), yes. Asset creation is not possible through the UI, and will only happen in an admin script. In theory someone could submit a new Asset directly to the API with the name `*&--!':screw)your?url.....`

Reply:
Looks like Mithril handles converting URL encoding on its own most places. I've added url encoding to this link though.